doc/source/ray-core/handling-dependencies.rst

```diff
@@ -336,64 +347,74 @@ And then a `test.py` like the following:
-
   @ray.remote
-  def message(entity):
-      import emoji
-      return emoji.emojize(entity + " rocks :thumbs_up:")
-
-  entities = [
-    "Ray",
-    "Ray Serve",
-    "Ray Data",
-    "Ray Train",
-    "Ray RLlib",
-    "Ray Tune",
-  ]
-  results = [message.remote(entity) for entity in entities]
-  for result in results:
-      print(ray.get(result))
+  def f():
+      return emoji.emojize('Python is :thumbs_up:')
+
+  # Execute 1000 copies of f across a cluster.
+  print(ray.get([f.remote() for _ in range(1000)]))
 
 
-and run the driver script with `uv run test.py`. For reproducibility it's
-recommended to freeze the package versions by running `uv lock`.
+and run the driver script with `uv run test.py`. This will run 1000 copies of
+the `f` function across a number of Python worker processes in a Ray cluster.
+The `emoji` dependency, in addition to being available for the main script, will
+also be available for all worker processes. Also, the source code in the current
+working directory will be available to all the workers.
 
 This workflow also supports editable packages, for example, you can use
 `uv add --editable ./path/to/package` where `./path/to/package`
-must be inside your `working_dir` so it's available to all
+must be inside your current working directory so it's available to all
 workers.
 
-Instead of the `pyproject.toml` file, you can also use a `requirements.txt`
-file and use `uv run --with-requirements requirements.txt` for your `py_executable`
-or use the `--with` flag to specify individual requirements.
+See `here <https://www.anyscale.com/blog/uv-ray-pain-free-python-dependencies-in-clusters#end-to-end-example-for-using-uv>`_
+for an end-to-end example of how to use `uv run` to run a batch inference workload
+with Ray Data.
 
+**Using uv in a Ray Job:** With the same `pyproject.toml` and `test.py` files as above,
+you can submit a Ray Job via
 
-.. tip::
+.. code-block:: sh
 
-  In order to make this functionality available in a convenient way without having
-  to specify `py_executable` in the runtime environment, you can use the following
-  runtime environment hook:
+  ray job submit --working-dir . -- uv run test.py
 
-  .. code-block:: sh
+This will make sure both the driver and workers of the job run in the uv environment as specified by your `pyproject.toml`.
 
-    export RAY_RUNTIME_ENV_HOOK=ray._private.runtime_env.uv_runtime_env_hook.hook
+**Using uv with Ray Serve:** With an appropriate `pyproject.toml` and `app.py` files, you can
+run a Ray Serve application with `uv run serve run app:main`.
+
+**Best Practices and Tips:**
+
+- Use `uv lock` to generate a lockfile and make sure all your dependencies are frozen, so things won't change in uncontrolled ways if a new version of a package gets released.
+
+- If you have a requirements.txt file, you can use `uv add -r requirement.txt` to add the dependencies to your `project.toml` and then use that with uv run.
```

Comment:
should this be pyproject.toml?

Reply:
Yes, I fixed it, thanks a lot for catching that!